packages/server/billing/helpers/generateInvoice.ts

```diff
@@ -182,9 +182,9 @@ const makeDetailedLineItems = async (itemDict: ItemDict, dataLoader: DataLoaderW
 
   for (let i = 0; i < userIds.length; i++) {
     // for each userId
-    const userId = userIds[i]
-    const email = emailLookup[userId]
-    const typesDict = itemDict[userId]
+    const userId = userIds[i]!
```

Comment:
+1 NIT: could refactor these for loops to `forEach` to improve readability and remove lines like this. I imagine you went with this approach for speed

Reply:
the forEach would only work on the userId array, I'd still have to go into 2 other lookup tables that could have an OOB index so i figured if it ain't broke 🤷